### DISCUSSION TITLE
Can you only access Fantasy Football League data from Seasons you were in?

### DISCUSSION THREAD

**Discussion by Jfreundlich01** [cwendt94/espn-api#525] [ESPN API] [answered]
Question may state the obvious. Have been using this api for a while and its been pretty smooth besides a few hiccups. Really appreciate all the work that has gone in to this! 

Question may state the obvious, but this past year I joined a new league. The league started in 2018. I can access this current years data. But anything starting with 2022 and beyond, I get a `League {league_id} cannot be accessed with espn_S2={espn_S2} and swid={swid} error message` 

Anyone else ran into this before or know a definite answer? Thanks!

**Comment by DesiPilla** [comment, marked as answer]
To my knowledge, public leagues can be accessed by anyone, regardless of whether they were in the league in or not (as you don't even need a `SWID` or `espn_s2` for public leagues).

However, for private leagues, yes you must have been a part of the league year in order to see the data. On ESPN's website, will it let you see older years from before you joined in the League History tab? I would assume no.

The only way to get this data is to get the `swid` and `espn_s2` of someone else who was in the league at the time.

**Reply by Jfreundlich01** [reply]
Ahhh yea, I can't see older years data on the history tab. That is tragic. Thanks!
